llvm/lib/Passes/PassBuilder.cpp

```diff
@@ -531,6 +532,12 @@ PassBuilder::PassBuilder(TargetMachine *TM, PipelineTuningOptions PTO,
                                           PARAMS)                              \
   PIC->addClassToPassName(CLASS, NAME);
 #include "llvm/Passes/MachinePassRegistry.def"
+
+      PIC->addClassToPassName(AsmPrinterInitializePass::name(),
```

Comment:
Does this work with the target-name-prefix-asm-printer like the legacy PM recently started handling?

Reply:
It doesn't work. There are two choices to compatible with legacy PM:
- Construct pass name by `getTargetTriple().getArchName()+"-asm-printer"` and use this name in `addClassToPassName`
- Let target add asm printer, then targets could provide `<Target>AsmPrinterPass` and add it to `<Target>PassRegistry.def`. This can keep the pass name and class name faithfully.